src/planscape/plan/views.py

```diff
@@ -408,6 +408,52 @@ def delete_projects(request: HttpRequest) -> HttpResponse:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
 
+@csrf_exempt
+def create_project_areas_for_project(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        owner = get_user(request)
+
+        body = json.loads(request.body)
+
+        # Get the project_id. This may come from an existing project or a
+        # placeholder project created for 1 forsys with patchmax run)
+        project_id = body.get('project_id', None)
+        if project_id is None or not (isinstance(project_id, int)):
+            raise ValueError("Must specify project_id as an integer")
+
+        # Get the Project, and if the user is logged in, make sure either
+        # 1. the Project owner and the owner are both None, or
+        # 2. the Project owner and the owner are both not None, and are equal.
+        project = Project.objects.get(pk=int(project_id))
+        if not ((owner is None and project.owner is None) or
```

Comment:
nit: it looks like this if statement is also used in create_project_areas. Could it be generalized in a separate function?

Reply:
eventually i'd like to deprecate one of these two calls (you can create a single project area by just populating the list with 1 item). let me add a TODO for that instead.